lib/services/role.go

```diff
@@ -920,6 +879,30 @@ func (r *RoleV2) V3() *RoleV3 {
 		role.Spec.Options[ForwardAgent] = true
 	}
 
+	// translate old v2 resources to v3 rules
+	rules := make(map[string][]string)
+	for resource, actions := range r.GetResources() {
+		var verbs []string
+
+		containsRead := utils.SliceContainsStr(actions, ActionRead)
+		containsWrite := utils.SliceContainsStr(actions, ActionWrite)
+
+		if containsRead && containsWrite {
+			verbs = RW()
+		} else if containsRead {
+			verbs = RO()
+		} else { // write-only
```

Comment:
again, "write" access becomes default behavior. what if there's a record with an empty string? it won't contain "write". I would make this explicit: grant RO() only if containsWrite.